hs/src/Reach/Eval/Core.hs

```diff
@@ -5034,6 +5102,15 @@ getReturnAnnot = \case
           JSReturn a _ _:_ -> Just a
           _ -> Nothing
 
+deconstructFun :: JSExpression -> App ([JSExpression], JSBlock)
+deconstructFun = \case
+  a@(JSArrowExpression _ _ cb) -> return $ (getFormals a, jsArrowBodyToRetBlock cb)
```

Comment:
You should make this robust to having parens and to using `function` and thus it should probably be a helper function... there might even already be something like in JSUtil